src/utils/media.ts

```diff
@@ -7,31 +8,31 @@ type DeprecatedPlaylistItem = {
   seriesPlaylistId?: string;
 };
 
-export const getSeriesId = (item: PlaylistItem & DeprecatedPlaylistItem) => {
-  if (!item) {
-    return undefined;
-  }
-
-  return item['seriesPlayListId'] || item.seriesPlaylistId || item.seriesId;
-};
-
 export const isPlaylist = (item: unknown): item is Playlist => !!item && typeof item === 'object' && 'feedid' in item;
 export const isPlaylistItem = (item: unknown): item is PlaylistItem => !!item && typeof item === 'object' && 'mediaid' in item;
 
-export const isSeriesPlaceholder = (item: PlaylistItem) => {
-  return typeof getSeriesId(item) !== 'undefined';
+// For the deprecated series flow we store seriesId in custom params
+export const getSeriesPlaylistIdFromCustomParams = (item: (PlaylistItem & DeprecatedPlaylistItem) | undefined) =>
+  item ? item.seriesPlayListId || item.seriesPlaylistId || item.seriesId : undefined;
+
+// For the deprecated flow we store seriesId in the media custom params
+export const isDeprecatedSeriesFlow = (item: PlaylistItem) => {
+  return typeof getSeriesPlaylistIdFromCustomParams(item) !== 'undefined';
 };
 
+// For the new series flow we use contentType custom param to define media item to be series
+// In this case media item and series item have the same id
+export const isNewSeriesFlow = (item: PlaylistItem) => item.contentType === CONTENT_TYPE.series;
+
+export const isSeries = (item: PlaylistItem) => isDeprecatedSeriesFlow(item) || isNewSeriesFlow(item);
```

Comment:
Naming adjustment and switch the order of legacy / new checks
```suggestion
export const isSeriesContentType = (item: PlaylistItem) => item.contentType === CONTENT_TYPE.series;

export const isSeries = (item: PlaylistItem) => isSeriesContentType(item) || isDeprecatedSeriesFlow(item);
```